exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/metadata/FileMetadataCollector.java

```diff
@@ -0,0 +1,278 @@
+    if (rowGroupMetadataList.isEmpty()) {
+      List<Metadata_V4.ColumnMetadata_v4> columnMetadataList = new ArrayList<>();
+      for (ColumnDescriptor columnDescriptor : schema.getColumns()) {
+
+        Statistics<?> stats = Statistics.getBuilderForReading(columnDescriptor.getPrimitiveType())
+          .withMax(null)
+          .withMin(null)
+          .withNumNulls(0)
+          .build();
+
+        addColumnMetadata(columnDescriptor.getPath(), stats,
+          columnDescriptor.getPrimitiveType().getPrimitiveTypeName(), columnMetadataList);
+      }
+
+      Metadata_V4.RowGroupMetadata_v4 rowGroupMeta = new Metadata_V4.RowGroupMetadata_v4(0L, 0L,
+        0L, getHostAffinity(0, 0L), columnMetadataList);
+      rowGroupMetadataList.add(rowGroupMeta);
+    }
+
+    Path path = Path.getPathWithoutSchemeAndAuthority(file.getPath());
+    Metadata_V4.ParquetFileMetadata_v4 parquetFileMetadata_v4 = new Metadata_V4.ParquetFileMetadata_v4(path, file.getLen(), rowGroupMetadataList);
+    this.fileMetadata = new Metadata_V4.ParquetFileAndRowCountMetadata(parquetFileMetadata_v4, totalNullCountMap, totalRowCount);
+  }
+
+  private void addColumnMetadata(String[] columnName,
+                                 Statistics<?> stats,
+                                 PrimitiveType.PrimitiveTypeName primitiveTypeName,
+                                 List<Metadata_V4.ColumnMetadata_v4> columnMetadataList) {
+    SchemaPath columnSchemaName = SchemaPath.getCompoundPath(columnName);
+    boolean thisColumnIsInteresting = allColumnsInteresting || columnSet == null || columnSet.contains(columnSchemaName.getRootSegmentPath());
+
+    if (skipNonInteresting && !thisColumnIsInteresting) {
+      return;
+    }
+
+    ColTypeInfo colTypeInfo = colTypeInfoMap.get(columnSchemaName);
+    long totalNullCount = stats.getNumNulls();
+    Metadata_V4.ColumnTypeMetadata_v4 columnTypeMetadata = new Metadata_V4.ColumnTypeMetadata_v4(
+      columnName, primitiveTypeName,
+      colTypeInfo.originalType, colTypeInfo.precision, colTypeInfo.scale,
+      colTypeInfo.repetitionLevel, colTypeInfo.definitionLevel, 0, false);
+    Metadata_V4.ColumnTypeMetadata_v4.Key columnTypeMetadataKey = new Metadata_V4.ColumnTypeMetadata_v4.Key(columnTypeMetadata.name);
+
+    totalNullCountMap.putIfAbsent(columnTypeMetadataKey, Metadata.DEFAULT_NULL_COUNT);
+    if (totalNullCountMap.get(columnTypeMetadataKey) < 0 || totalNullCount < 0) {
+      totalNullCountMap.put(columnTypeMetadataKey, Metadata.NULL_COUNT_NOT_EXISTS);
+    } else {
+      long nullCount = totalNullCountMap.get(columnTypeMetadataKey) + totalNullCount;
+      totalNullCountMap.put(columnTypeMetadataKey, nullCount);
+    }
+    if (thisColumnIsInteresting) {
+      // Save the column schema info. We'll merge it into one list
+      Object minValue = null;
+      Object maxValue = null;
+      boolean statsAvailable = !stats.isEmpty();
+      if (statsAvailable) {
+        if (stats.hasNonNullValue()) {
```

Comment:
Could you please combine these two if statements?

P.S. Since this code was moved from another place, it is optional to address this comment, but in either way, I've written it :)